src/reconcile/rec_visibility.c

```diff
@@ -414,12 +414,47 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
             /* Find the update this tombstone applies to. */
             if (!__wt_txn_upd_visible_all(session, upd)) {
-                while (upd->next != NULL && upd->next->txnid == WT_TXN_ABORTED)
-                    upd = upd->next;
+                /*
+                 * Loop until a valid update from a different transaction is found in the update
+                 * list.
+                 */
+                while (upd->next != NULL) {
+                    if (upd->next->txnid == WT_TXN_ABORTED)
+                        upd = upd->next;
+                    else if (tombstone->start_ts == upd->next->start_ts &&
+                      tombstone->durable_ts == upd->next->durable_ts &&
+                      tombstone->txnid == upd->next->txnid) {
```

Comment:
Can we compare txnid before timestamps?

Reply:
Can we add an assert to ensure same_txn_valid_upd is not a tombstone?

Reply:
Fixed. And added asserts in the place where the `same_txn_valid_upd` is used.